File: client/src/main/java/xyz/chunkstories/client/net/packets/ClientPacketWorldUser.java
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.client.net.packets;

import java.io.DataInputStream;
import java.io.IOException;

import xyz.chunkstories.api.exceptions.net.IllegalPacketException;
import xyz.chunkstories.api.net.PacketReceptionContext;
import xyz.chunkstories.net.packets.PacketWorldUser;
import xyz.chunkstories.api.world.World;
import xyz.chunkstories.client.ingame.LocalPlayerImplementation;
import xyz.chunkstories.client.net.ClientPacketsEncoderDecoder;

public class ClientPacketWorldUser extends PacketWorldUser {

	public ClientPacketWorldUser(World world) {
		super(world);
	}

	@Override
	public void process(PacketSender sender, DataInputStream in, PacketReceptionContext context) throws IOException {
		super.process(sender, in, context);

		System.out.println("weeiiird");
		if (context instanceof ClientPacketsEncoderDecoder) {
			try {
				((LocalPlayerImplementation) ((ClientPacketsEncoderDecoder) context).getContext().getPlayer()).getLoadingAgent()
						.handleServerResponse(this);
			} catch (IllegalPacketException e) {
				e.printStackTrace();
			}
		}
	}
}

File: client/src/main/java/xyz/chunkstories/client/net/packets/ClientPacketWorldUser.kt
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//
package xyz.chunkstories.client.net.packets

import xyz.chunkstories.api.client.Client
import xyz.chunkstories.net.packets.PacketWorldUser
import xyz.chunkstories.api.player.Player
import xyz.chunkstories.api.world.World
import xyz.chunkstories.client.ingame.IngameClientImplementation
import java.io.DataInputStream

class ClientPacketWorldUser(world: World) : PacketWorldUser(world) {
    override fun receive(dis: DataInputStream, player: Player?) {
        super.receive(dis, player)
        val instance = world.gameInstance
        if (instance is Client) {
            val ingame = instance.ingame ?: return
            (ingame as IngameClientImplementation).loadingAgent.handleServerResponse(this)
        }
    }
}